src/Microsoft.DotNet.Interactive.HttpRequestParser/Parser/HttpRequestParser.cs

```diff
@@ -322,30 +325,37 @@ private void ConsumeCurrentTokenInto(HttpSyntaxNode node)
             {
                 requestNode.Add(versionNode);
             }
-            else 
+            else
             {
-                ParseTrailingTrivia(requestNode, stopAfterNewLine: true);
+                ParseTrailingWhitespace(requestNode, stopAfterNewLine: true);
+            }
+
+            var commentsToPrepend = new List<HttpCommentNode>();
+            if (ParseComment() is { } comment)
```

Comment:
This might depend on whether a previous call parsed the trailing whitespace. 

For what it's worth, the old parser reports a diagnostic when there's whitespace on the line preceding the `#`.